app/src/androidTest/java/com/google/jetpackcamera/ImageCaptureDeviceTest.kt

```diff
@@ -95,25 +97,33 @@ internal class ImageCaptureDeviceTest {
 
     @Test
     fun image_capture_external_illegal_uri() = runTest {
-        val timeStamp = System.currentTimeMillis()
-        val inputUri = Uri.parse("asdfasdf")
-        getTestRegistry {
-            activityScenario = ActivityScenario.launchActivityForResult(it)
-            uiDevice.wait(
-                Until.findObject(By.res(CAPTURE_BUTTON)),
-                5000
-            )
-            uiDevice.findObject(By.res(CAPTURE_BUTTON)).click()
-            uiDevice.wait(
-                Until.findObject(By.res("ImageCaptureFailureToast")),
-                5000
+        val uri = Uri.parse("asdfasdf")
+        activityScenario = ActivityScenario.launch(getExternalIntent(uri))
+
+        composeTestRule.waitUntil(timeoutMillis = APP_START_TIMEOUT_MILLIS) {
+            composeTestRule.onNodeWithTag(CAPTURE_BUTTON).isDisplayed()
+        }
+        composeTestRule.onNodeWithTag(CAPTURE_BUTTON)
+            .assertExists()
+            .performClick()
+        composeTestRule.waitUntil(timeoutMillis = APP_START_TIMEOUT_MILLIS) {
+            composeTestRule.onNodeWithTag(IMAGE_CAPTURE_FAIL_TOAST).isDisplayed()
+        }
+    }
+
+    private fun getExternalIntent(uri: Uri): Intent {
+        val intent = Intent(Intent.ACTION_MAIN)
+        intent.setAction(MediaStore.ACTION_IMAGE_CAPTURE)
+        intent.setComponent(
+            ComponentName(
+                "com.google.jetpackcamera",
+                "com.google.jetpackcamera.MainActivity"
             )
-            uiDevice.pressBack()
-            activityScenario!!.result
-        }.register("key_illegal_uri", TEST_CONTRACT) { result ->
-            assert(!result)
-        }.launch(inputUri)
-        deleteFilesInDirAfterTimestamp(timeStamp)
+        )
+
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, uri)
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
```

Comment:
I don't think we want to use this flag. We should ensure the app is closed from the previous test before running the next test instead. This should be possible by ensuring `ActivityScenario.close()` is called at the end of each test. This can be accomplished using `use`, which is what `runScenarioTest` does in our `UiTestUtil.kt` file.